src/main/java/io/dexi/config/DexiConfig.java

```diff
@@ -0,0 +1,235 @@
+        T configuration = null;
+
+        Parameters parameters = new Parameters();
+        PropertiesBuilderParameters properties = parameters.properties();
+        FileBasedConfigurationBuilder<T> builder = new ReloadingFileBasedConfigurationBuilder<>(filedBasedClazz);
+
+        URI uri = new URI(fileLocation);
+        if (isUrl(fileLocation)) {
+            builder = builder.configure(properties.setURL(uri.toURL()));
+            configuration = builder.getConfiguration();
+        } else {
+            URL localFileURL = DexiConfig.class.getResource(fileLocation);
+            if (localFileURL != null) {
+                builder = builder.configure(properties.setFileName(localFileURL.getFile()));
+                configuration = builder.getConfiguration();
+            }
+        }
+
+        return configuration;
+    }
+
+    private static void addConfigurationToProperties(Configuration configuration) {
+        if (configuration != null) {
+            Iterator<String> keys = configuration.getKeys();
+            while (keys.hasNext()) {
+                String keyWithSection = keys.next();
+
+                String existingValue = properties.getProperty(keyWithSection);
+                if (existingValue == null) {
+                    String value = configuration.getString(keyWithSection);
+                    properties.put(keyWithSection, value);
+                }
+            }
+        }
+    }
+
+    private static void getConfigurationFromURL(String url) throws MalformedURLException, ConfigurationException, URISyntaxException {
+        Configuration ymlConfigurationURL = getConfigurationFile(url, YAMLConfiguration.class);
+        addConfigurationToProperties(ymlConfigurationURL);
+    }
+
+    private static void readLocalConfiguration(String fileLocation) throws MalformedURLException, ConfigurationException, URISyntaxException {
+        String fileExtension = fileLocation.substring(fileLocation.lastIndexOf(".") + 1);
+
+        Class<? extends FileBasedConfiguration> configurationClass;
+        switch (fileExtension) {
+            case "yml":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            case "json":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            case "xml":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            case "ini":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported file extension " + fileExtension);
+        }
+
+        Configuration ymlConfigurationLocal = getConfigurationFile(fileLocation, configurationClass);
+        addConfigurationToProperties(ymlConfigurationLocal);
+    }
+
+    private static boolean isUrl(String fileLocation) {
+        final Pattern urlPattern = Pattern.compile("https?://");
```

Comment:
No need to change but usually compiled Pattern's go into a private static final variable since it never changes and it does cost a little bit to compile a regex

Reply:
It doesn't matter here since this method is not being called in a large loop or similar